scim/client.bal

```diff
@@ -0,0 +1,262 @@
+    # + clientId - The client ID of the application
+    # + clientSecret - The client secret of the application
+    # + scope - permitted scopes
+    public isolated function init(string orgName, string clientId, string clientSecret, string[] scope) returns error? {
+        self.orgName = orgName;
+        self.clientId = clientId;
+        self.clientSecret = clientSecret;
+        self.scope = scope;
+
+        self.provider = new ({
+            tokenUrl: HOST_URL + TENANT_PATH + self.orgName + "/oauth2/token",
+            clientId: clientId,
+            clientSecret: clientSecret,
+            scopes: scope
+        });
+
+        self.clientEndpoint = check new (HOST_URL + TENANT_PATH + self.orgName + SCIM, {httpVersion: "1.1"});
+
+        string token = check self.provider.generateToken();
+        io:print(token);
+
+        self.headers = {
+            "accept": "application/scim+json",
+            "Authorization": "Bearer " + token
+        };
+    }
+
+    # Gets the list of users.
+    #
+    # + return - The list of users
+    @display {label: "Get Users"}
+    public isolated function getUsers() returns UserResponse|error {
+        json|error response = self.clientEndpoint->get(USERS, self.headers, json);
+        if (response is json) {
+            UserResponse users = check response.cloneWithType(UserResponse);
+            return users;
+        } else {
+            return response;
+        }
+    }
+
+    # Gets a user by the user ID.
+    #
+    # + id - The ID of the user
+    # + return - The user
+    @display {label: "Get User by ID"}
+    public isolated function getUser(@display {label: "User Id"} string id) returns UserResource|error {
+        json|error response = self.clientEndpoint->get(USERS + "/" + id, self.headers, json);
+        if (response is json) {
+            UserResource user = check response.cloneWithType(UserResource);
+            return user;
+        } else {
+            return response;
+        }
+    }
+
+    # Creates a user.
+    #
+    # + data - The user data
+    # + return - The created user
+    @display {label: "Create User"}
+    public isolated function createUser(@display {label: "User data"} SCIMUser data) returns UserResource|error {
+        json|error response = self.clientEndpoint->post(USERS, data, self.headers, (), json);
+        if (response is json) {
+            UserResource user = check response.cloneWithType(UserResource);
+            return user;
+        } else {
+            return response;
+        }
+    }
+
+    # Updates a user.
+    #
+    # + id - The ID of the user
+    # + data - The user data to be updated
+    # + return - The updated user
+    @display {label: "Update User"}
+    public isolated function updateUser(@display {label: "User Id"} string id, @display {label: "User updated data"} UserUpdate data) returns UserResource|error {
+        json|error response = self.clientEndpoint->put(USERS + "/" + id, data, self.headers, (), json);
```

Comment:
Since you are just returning the error
```suggestion
        json response = check self.clientEndpoint->put(USERS + "/" + id, data, self.headers, (), json);
```